macros/contexts/contextNondecimalBase.pl

```diff
@@ -0,0 +1,273 @@
+
+=cut
+
+sub _contextNondecimalBase_init {
+	context::NondecimalBase::Init();
+	sub setBase     { context::NondecimalBase::setBase(@_); }
+	sub convertBase { context::NondecimalBase::convert(@_); }
+}
+
+###########################################################################
+
+package context::NondecimalBase;
+
+# defines the base for the context.
+our $base = 10;
+#
+#  The standard digits, pre-built so it doesn't have to be done each time the conversion is called
+#
+our $digits16 = [ '0' .. '9', 'A' .. 'F' ];
+our $digit16  = { map { ($digits16->[$_], $_) } (0 .. scalar(@$digits16) - 1) };
+
+use Data::Dumper;
+#
+#  Initialize the contexts and make the creator function.
+#
+sub Init {
+	my $context = $main::context{NondecimalBase} = Parser::Context->getCopy("Numeric");
+	$context->{name} = 'NondecimalBase';
+	$context->{parser}{Number} = 'context::NondecimalBase::Number';
+	# $context->{value}{Real} = 'context::Hex::Hex';
+	$context->{pattern}{number} = '[0-9A-F]+';
+	$context->functions->disable('All');
+	$context->operators->remove('^');
+	$context->parens->remove('|');
+	$context->constants->clear();
+	# $context->operators->add(
+	#   '&'  => {precedence => .6, associativity => 'left', type => 'bin', string => ' & ',
+	#            class => 'context::Hex::BOP::hex', eval => sub {$_[0] & $_[1]}},
+	#   '|'  => {precedence => .5, associativity => 'left', type => 'bin', string => ' | ',
+	#            class => 'context::Hex::BOP::hex', eval => sub {$_[0] | $_[1]}},
+	#   '^'  => {precedence => .5, associativity => 'left', type => 'bin', string => ' ^ ',
+	#            class => 'context::Hex::BOP::hex', eval => sub {$_[0] ^ $_[1]}},
+	#   '>>' => {precedence => .4, associativity => 'left', type => 'bin', string => ' >> ',
+	#            class => 'context::Hex::BOP::hex', eval => sub {$_[0] >> $_[1]}},
+	#   '<<' => {precedence => .4, associativity => 'left', type => 'bin', string => ' << ',
+	#            class => 'context::Hex::BOP::hex', eval => sub {$_[0] << $_[1]}},
+	#   '~'  => {precedence => 6, associativity => 'left', type => 'unary', string => '~',
+	#            class => 'context::Hex::UOP::not'},
+	# );
+	# $context->{precedence}{Hex} = $context->{precedence}{special};
+	$context->flags->set(limits => [ -1000, 1000, 1 ]);
+	$context->update;
+
+	# main::PG_restricted_eval('sub Hex {context::Hex::Hex->new(@_)}');
+}
+use Data::Dumper;
+
+sub convert {
+	my $value = shift;
+	print Dumper "convert: $value";
+	# Set default options and get passed in options.
+	my %options = (
+		from   => 10,
+		to     => 10,
+		digits => $digits16,
+		@_
+	);
+	my $from   = $options{'from'};
+	my $to     = $options{'to'};
+	my $digits = $options{'digits'};
+	print Dumper "from: $from; to: $to";
+
+	die "The digits option must be an array of characters to use for the digits"
+		unless ref($digits) eq 'ARRAY';
+
+	#
+	# The highest base the digits will support
+	#
+	my $maxBase = scalar(@$digits16);
```

Comment:
Since you are using `$digits` as the digits, you need to use `@$digits` here, not `@$digits16`.  When not specified by the user, `$digits` will be `$digits16`, but if they use `digits => ['0', '1']`, for example, you need to restrict them to base 2, while if they use `digits => ['0' .. '9', 'A' .. 'Z']`, they can use bases higher than 16.